Core/src/main/java/com/plotsquared/core/services/plots/AutoQuery.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * PlotSquared, a land and world management plugin for Minecraft.
+ * Copyright (C) IntellectualSites <https://intellectualsites.com>
+ * Copyright (C) IntellectualSites team and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package com.plotsquared.core.services.plots;
+
+import com.plotsquared.core.player.PlotPlayer;
+import com.plotsquared.core.plot.PlotArea;
+import com.plotsquared.core.plot.PlotId;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public record AutoQuery(
+        PlotPlayer<?> player,
+        PlotId startId,
+        int sizeX,
+        int sizeZ,
+        PlotArea plotArea
+) {
+
+    /**
+     * Crate a new auto query
+     *
+     * @param player   Player to claim for
+     * @param startId  Plot ID to start searching from
+     * @param sizeX    Number of plots along the X axis
+     * @param sizeZ    Number of plots along the Z axis
+     * @param plotArea Plot area to search in
+     */
+    public AutoQuery(
```

Comment:
Annotations can be added to the record components, and the `@param` tag is allowed at the record header too. Can we remove the constructor as well as the methods?